src/app/accounts/add-earn.ts

```diff
@@ -33,6 +39,13 @@ export const addEarn = async ({
   if (validatedPhoneMetadata instanceof Error)
     return new InvalidPhoneMetadataForRewardError(validatedPhoneMetadata.name)
 
+  const userIP = await UsersIpRepository().findById(user.id)
```

Comment:
the user is already be prefetched during the authentication, and could be passed from graphql layer to the app layer. this would avoid another request to mongodb. you can look for `domainUser` to see some other example pattern.

Reply:
oh nevermind, we don't fetch user here but userIp

Reply:
Taking a look at `src/services/mongoose/users.ts`, the `lastIPs` property on the `user` object is not returned in the result set when fetching a user from the graphql layer. 

I implemented it this way as I assumed there was a reason for not wanting to fetch `lastIPs` in `Users.getUserForLogin`. 

Just want to confirm with you that you would like me to start returning  the `lastIPs` property on the user object in the graphql layer and pass it down, as opposed to fetching it as needed. 

Happy to implement this way, just wanted to check with you first to make sure.